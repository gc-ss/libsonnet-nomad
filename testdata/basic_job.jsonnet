local nomad = import '../nomad.libsonnet';

nomad.Job('some-job', {
  groups: [
    nomad.Group('some-group', {
      tasks: [
        nomad.DockerTask {
          name: 'some-task',
          image: 'some-image',
        },
      ],
    }),
  ],
})
