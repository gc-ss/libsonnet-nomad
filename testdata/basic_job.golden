{
   "Job": {
      "Datacenters": [
         "dc1"
      ],
      "ID": "some-job",
      "TaskGroups": [
         {
            "Name": "some-group",
            "Networks": [ ],
            "Services": [
               {
                  "Name": "some-service",
                  "PortLabel": 9001,
                  "Tags": [ ]
               }
            ],
            "Tasks": [
               {
                  "Config": {
                     "image": "some-image",
                     "network_mode": "host"
                  },
                  "Driver": "docker",
                  "Name": "some-task",
                  "Services": [ ],
                  "Templates": [ ]
               }
            ]
         }
      ]
   }
}
